src/VegaWrapper2.js

```diff
@@ -0,0 +1,398 @@
+const makeValidator = require('domain-validator'),
+      parseWikidataValue = require('wd-type-parser');
+
+function validate(urlObj, name, min, max, isFloat) {
+    let value = urlObj[name];
+    if (value === undefined) {
+        throw new Error(urlObj.type + ': parameter ' + name + ' is not set');
+    }
+    if (!(isFloat ? /^-?[0-9]+\.?[0-9]*$/ : /^-?[0-9]+$/).test(value)) {
+        throw new Error(urlObj.type + ': parameter ' + name + ' is not a number');
+    }
+    value = isFloat ? parseFloat(value) : parseInt(value);
+    if (value < min || value > max) {
+        throw new Error(urlObj.type + ': parameter ' + name + ' is not valid');
+    }
+}
+
+class VegaWrapper2 {
+    /**
+     * Shared library to wrap around vega code
+     * @param {Object} wrapperOpts Configuration options
+     * @param {Object} wrapperOpts.loader Vega-loader object, its sanitize(), file() and load() method will be overwrited
+     * @param {Object} wrapperOpts.domains allowed protocols and a list of their domains
+     * @param {Object} wrapperOpts.domainMap domain remapping
+     * @param {Function} wrapperOpts.logger
+     * @param {Function} wrapperOpts.formatUrl
+     * @param {string} [wrapperOpts.languageCode]
+     * @constructor
+     */
+    constructor(wrapperOpts) {
+        // Copy all options into the wrapper
+        Object.assign(this, wrapperOpts);
+        this.validators = {};
+
+        this.loader.sanitize = this.sanitize.bind(this);
+        this.loader.load = (uri, options) => {
```

Comment:
This whole block can be shortened like this. Also, there is no need to nest `.then()`, instead they can all be called on the same level, like this:
```suggestion
        this.loader.load = (uri, options) => 
                this.sanitize(uri, options)
                .then(opt => this.loader.http(opt.href, options))
                .then(txt => this.parseResponse(txt, uri.type, options));
```